fhir/com.b2international.snowowl.fhir.core/src/com/b2international/snowowl/fhir/core/request/conceptmap/FhirWriteSupport.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2022 B2i Healthcare Pte Ltd, http://b2i.sg
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.b2international.snowowl.fhir.core.request.conceptmap;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import com.b2international.snowowl.core.ResourceURI;
+import com.b2international.snowowl.core.ServiceProvider;
+import com.b2international.snowowl.core.request.ResourceRequests;
+import com.b2international.snowowl.fhir.core.exceptions.BadRequestException;
+
+/**
+ * @since 8.7.1
+ */
+public interface FhirWriteSupport {
+
+	/**
+	 * Checks whether a business version can be created based on an incoming resource snapshot state (FHIR Resource representation).
+	 * 
+	 * @param context
+	 * @param resourceUri
+	 * @param newVersionToCreate
+	 * @param newVersionEffectiveDate
+	 * @return a boolean value indicating whether the version can be created either forcefully or via the standard way.
+	 * @throws BadRequestException - if the version creation cannot be completed for any reason
+	 */
+	default boolean checkBusinessVersion(
+			final ServiceProvider context, 
+			final ResourceURI resourceUri, 
+			final String newVersionToCreate, 
+			final LocalDate newVersionEffectiveDate) {
+			
+		return ResourceRequests.prepareSearchVersion()
+			.setLimit(1)
+			.filterByResource(resourceUri)
+			.sortBy("effectiveTime:desc")
```

Comment:
It would be nice to replace this with a field reference: 
```suggestion
			.sortBy(Sort.fieldDesc(VersionDocument.Fields.EFFECTIVE_TIME))
```

Reply:
Not sure about this. We usually just use the String value as to make it simple to read and understand how a result set is being sorted. We can have a quick chat about this.